### PR TITLE
apply pr feedback & and include second review findings

### PR DESCRIPTION
**Pull request by lstreckeisen**
Implements my feedback on #128 and my review findings up to section 7

### CODE REVIEW

File: SA/design_concept/content/design/design_iteration_2_decisions_application.typ [hunk 1]
```diff
@@ -2,13 +2,9 @@
 #import "@preview/tablex:0.0.6": tablex, hlinex
 #let load-drawio-svg(path, ..args) = image.decode(read(path).replace("Text is not SVG - cannot display", ""), ..args)
 
-= Application <application-decision>
+= Function Application <application-decision>
 
-The design suggested after the second iteration, as described in
-@design-proposal-2, leaves the application style open as a decision.
-This section makes that decision and elaborates on the reasons.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_iteration_2_decisions_application.typ:9]: Why would you like to remove this paragraph? There reference to this section is still there

Comment (fl3pp) [reply SA/design_concept/content/design/design_iteration_2_decisions_application.typ:9]: ah, I see, you removed them as well.

Wouldn't we need to add another paragraph instead to give more context?

File: SA/design_concept/content/design/functional_requirements.typ [hunk 2]
```diff
@@ -37,38 +37,42 @@ By default, "user" in the use case description refers to the "student user".
     caption: "Use Case Diagram"
 )<use_case_diagram>
 
+As the aim of this project is to find a visual representation of functional programming,
+the use case descriptions are kept very brief and only state the intention behind the use case but not how it must be achieved.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:41]:
```suggestion
the use case descriptions are kept brief and only state the intention behind the use case, not how it must be achieved.
```

Not sure about the last sentence. I think a use case should never state how it is to be achieved

File: SA/design_concept/content/design/functional_requirements.typ [hunk 3]
```diff
@@ -37,38 +37,42 @@ By default, "user" in the use case description refers to the "student user".
...
+
 === UC1 - Simple Function Composition
-A user wants to compose a simple function using pre-defined functions, e.g., Integer parameters.
+A user wants to execute their visually composed functions to see the effects of their functions on data.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:44]: Is this the right UC? Or did something get mixed up?

File: SA/design_concept/content/design/functional_requirements.typ [hunk 4]
```diff
@@ -37,38 +37,42 @@ By default, "user" in the use case description refers to the "student user".
...
 
 === UC2 - Function Execution
 A user wants to execute their visually composed functions.
 
 === UC3 - Recursive Function Composition
-A user wants to compose a function that is defined by itself.
+A user wants to compose a function that is defined using itself.
+For that, the user needs possibilities to distinguish between a recursive and a base case.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:51]:
```suggestion
To do so, the user needs possibilities to distinguish between a recursive and a base case.
```

File: SA/design_concept/content/design/functional_requirements.typ [hunk 5]
```diff
@@ -37,38 +37,42 @@ By default, "user" in the use case description refers to the "student user".
...
 
 === UC4 - Function Composition using Higher-Order Functions
-A user wants to compose functions that take other functions as their argument.
+To create generic functions that can be used in different contexts, a user wants to compose functions that take other functions as their argument.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:54]: not sure whether the first sentence provides additional value. I would be fine with leaving it out

**Review by fl3pp — APPROVED** [review]

File: SA/design_concept/content/design/design_iteration_2_decisions_application.typ (hunk 1 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/design_iteration_2_decisions_application.typ:9]: I think its ok. Like this, the sum type destruction and function application sections are just part of the concept description.

File: SA/design_concept/content/design/functional_requirements.typ (hunk 2 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ:41]: Not in detail, but I think descriptions like 'click on a button', etc. are sometimes included. But I can also shorten the sentence.

Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ]: I took the to be achieved part out.

File: SA/design_concept/content/design/functional_requirements.typ (hunk 3 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ]: Moved to the right place

File: SA/design_concept/content/design/design_iteration_2_decisions_application.typ (hunk 1 above)
Comment (fl3pp) [reply SA/design_concept/content/design/design_iteration_2_decisions_application.typ:9]: ok 👍

File: SA/design_concept/content/design/functional_requirements.typ (hunk 5 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ:54]: I tried to provide the intention behind using hight-order functions

Comment (fl3pp) [reply SA/design_concept/content/design/functional_requirements.typ:54]: What about:
```suggestion
To create reusable and composable functions, a user wants to compose functions that take other functions as their input, in other words, higher-order functions.
```

**Pull request merged**